pipeline/pipelines/core_generation_pipeline.py

```diff
@@ -55,13 +55,10 @@ class _GoCoreTaskFactory(code_gen.TaskFactoryBase):
     """
 
     def get_tasks(self, **kwargs):
-        return [
-            protoc_tasks.ProtoCodeGenTask('CoreProtoGen',
-                                          inject=kwargs),
-            protoc_tasks.GoExtractImportBaseTask('ExtractGoPackageName',
-                                                 inject=kwargs),
-            protoc_tasks.GoLangUpdateImportsTask('UpdateImports',
-                                                 inject=kwargs)]
+        tasks = [protoc_tasks.ProtoCodeGenTask,
+                 protoc_tasks.GoExtractImportBaseTask,
+                 protoc_tasks.GoLangUpdateImportsTask]
+        return task_utils.instantiate_tasks(tasks, kwargs)
```

Comment:
I guess I am not really seeing the benefits of this approach, or how it would work for our batch pipeline. Consider the situation after we have added a task that creates a PR after generating all of the api x language combinations. Our list might look like this:

`[PrepGoogleApisDir, [[Gapic1-java-pubsub, Gapic2-java-pubsub], [Gapic1-java-logging, Gapic2-java-logging]], CreatePR]`

So somehow we need to encode that the top-level list needs to be serial, and each of the individual `[Gapic1, Gapic2]` lists needs to be serial, but that `[Gapic1-java-pubsub, Gapic2-java-pubsub]` can be done in parallel with `[Gapic1-java-logging, Gapic2-java-logging]`. Then add in the possibility that we might want to make some individual task like `Gapic1` actually be a flow (for example, gapic publishing could be a flow) and we are basically recreating the existing taskflow structure, just to avoid calling the instantiation function.